### ISSUE TITLE
Style/RedundantConditional should cover cases where only true is returned

### ISSUE THREAD

**Issue by Zopolis4** [rubocop/rubocop#13060]
## Is your feature request related to a problem? Please describe.

I was looking back at the code I had written and realised this was probably not a good way of doing it, and I think it should be covered as a rubocop offense.
```ruby
git = true unless @pkg.git_hashtag.to_s.empty?
```

## Describe the solution you'd like

For `Style/RedundantConditional` to be extended (and made unsafe) to cover cases where only true or false are returned, so:
```ruby
# bad
if x == y
  foo = true
end

# bad
foo = true if x == y

# good
foo = x == y
```

This would be unsafe in certain scenarios, where there is an explicit reliance on a variable being nil as opposed to the more common check for falsiness:
```ruby
# bad
bar = true if '1' == '2'

# good
bar = '1' == '2'

# code affected by this behaviour change
puts 'falsey' unless bar
puts 'nil' if bar.nil?
```

```
# output when not autocorrected:
falsey
nil
# output when autocorrected:
falsey
```

## Describe alternatives you've considered

The creation of a new cop to cover these cases to allow `Style/RedundantConditional` to remain safe to autocorrect, or maybe for `Style/RedundantConditional` to only do these particular autocorrects when an unsafe autocorrect is requested. 

## Additional context

I think this is relevant to `Style/RedundantConditional`? It might be under `Style/IfWithBooleanLiteralBranches`, I don't entirely understand.

**Comment by Zopolis4** [comment]
For feature requests, what is the protocol around them becoming stale? To my knowledge this feature has not been implemented, and I still would benefit from it, so is that enough for the issue to remain open?

**Comment by Zopolis4** [comment]
not stale

**Comment by Zopolis4** [comment]
not stale

**Comment by Zopolis4** [comment]
not stale

**Comment by Zopolis4** [comment]
not stale

**Comment by Zopolis4** [comment]
not stale

**Comment by augustocbx** [comment]
Hi, is there interest in moving this forward or should it be closed?

I checked on current master. Today `Style/RedundantConditional` only flags when both branches are boolean literals, like:

```ruby
if x == y
  true
else
  false
end
```

The cases in this issue stay silent:

```ruby
foo = true if x == y
bar = true if '1' == '2'
```

So the request is to make the cop detect these too. The catch is it would need to register new offenses and the autocorrect would have to be unsafe, since `bar` becomes `false` instead of `nil` and that can change behavior.

I'm happy to work on it, but I'd rather get a maintainer's call first: do you want this inside `Style/RedundantConditional`, or does it fit better in `Style/IfWithBooleanLiteralBranches`? Or is this out of scope and the issue should be closed?

Thanks.
